### PR TITLE
fix(TDI-39423):Amend the escape method.

### PR DESCRIPTION
**Pull request by pyzhou-talend**
https://jira.talendforge.org/browse/TDI-39423

### CODE REVIEW

**Comment by pyzhou-talend** [comment]
@pteyssier Because it still have the sequence issue by using replace all. i.e "content=c:\\note.txt" No matter what order I put "replaceAll("\\\\\\\\","\\\\")" & "replaceAll("\\\\n","\n")" , it won't get the correct result. And as you see it need to escape two times. It's hard for others to understand the code like  "replaceAll("\\\\\\\\","\\\\")"

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet [hunk 1]
```diff
@@ -1361,37 +1361,39 @@ if (execStat) {
...
-			// judege if the left string includes escape chars
-			for (String[] strArray : escapeChars) {
-				index = keyValue.indexOf(strArray[0],currIndex);
-				if (index>=0) {
-
-					result.append(keyValue.substring(currIndex, index + strArray[0].length()).replace(strArray[0], strArray[1]));
-					currIndex = index + strArray[0].length();
-					break;
+		int length = contentToReplace.length();
+		boolean flag = false;
```
Comment (pteyssier) [inline main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: Even better without flag:
1) remove this line

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet [hunk 2]
```diff
@@ -1361,37 +1361,39 @@ if (execStat) {
...
+
+		for (int i = 0; i < length; i++) {
+
+			if (contextChars[i] == '\\') {
+				for (char[] escapeChar : escapeChars) {
+					if (i + 1 < length && contextChars[i + 1] == escapeChar[0]) {
+						result.append(escapeChar[1]);
```
Comment (pteyssier) [inline main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: 2) Revove this line as it's equivalent to result.append(contextChars[i + 1]) which is equivalent to those 2 lines you already have below: 
i++;
...
result.append(contextChars[i])

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet [hunk 3]
```diff
@@ -1361,37 +1361,39 @@ if (execStat) {
...
+						i++;
+						flag = true;
```
Comment (pteyssier) [inline main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: 3) remove flag = true

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet [hunk 4]
```diff
@@ -1361,37 +1361,39 @@ if (execStat) {
...
+						break;
+					}
 				}
+
 			}
-			// if the left string doesn't include escape chars, append the left into the result
-			if (index < 0) {
-				result.append(keyValue.substring(currIndex));
-				currIndex = currIndex + keyValue.length();
+			if (!flag) {
```
Comment (pteyssier) [inline main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: remove the if (!flag)

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet [hunk 5]
```diff
@@ -1361,37 +1361,39 @@ if (execStat) {
...
+				result.append(contextChars[i]);
 			}
-		}
+			flag = false;
```
Comment (pteyssier) [inline main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: 4) remove the flag = false

File: main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet (hunk 2 above)
Comment (pteyssier) [reply main/plugins/org.talend.designer.codegen/jet_stub/footer.javajet]: Sorry I misread contextChars with escapeChar

**Pull request merged**